include/boost/geometry/views/closeable_view.hpp

```diff
@@ -44,19 +47,23 @@ struct closing_view
 {
     // Keep this explicit, important for nested views/ranges
     explicit inline closing_view(Range& r)
-        : m_range(r)
+        : m_range(boost::ref(r))
     {}
 
     typedef closing_iterator<Range> iterator;
     typedef closing_iterator<Range const> const_iterator;
 
-    inline const_iterator begin() const { return const_iterator(m_range); }
-    inline const_iterator end() const { return const_iterator(m_range, true); }
+    inline const_iterator begin() const { return const_iterator(cref()); }
+    inline const_iterator end() const { return const_iterator(cref(), true); }
+
+    inline iterator begin() { return iterator(ref()); }
+    inline iterator end() { return iterator(ref(), true); }
 
-    inline iterator begin() { return iterator(m_range); }
-    inline iterator end() { return iterator(m_range, true); }
 private :
-    Range& m_range;
+    inline Range& ref() { return m_range; }
+    inline Range const& cref() const { return m_range; }
+
+    boost::reference_wrapper<Range> m_range;
 };
 
```

Comment:
I understand the reasoning but this is a semantic change for Views in general. I'm not sure if this is a good idea to allow copying. Semantically this makes View concept closer to Range. Or in other words it changes something that had reference semantics into something having pointer semantics.

E.g. one could think about returning them from a function and this could be dangerous in some cases.

Maybe we should have some different concept for such Range-like weak Geometry pointing-"views"?
Or maybe do you see some different, relatively simple way of implementing it?